app/models/question.rb

```diff
@@ -1,6 +1,17 @@
 class Question < ActiveRecord::Base
 
   has_many :answers
-  belongs_to :user
+  belongs_to :user, inverse_of: :questions
+
+  def self.all_by_query_params(params)
+    questions = self.where(private: false)
+    if params[:question_name]
+      questions = questions.where('title LIKE ?', '%' + params[:question_name] + '%')
```

Comment:
I feel like there may be a cleaner way to do this but I'm not sure. Any thoughts?